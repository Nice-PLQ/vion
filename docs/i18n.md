# 国际化

Vion 默认采用中文语言，如果需要切换其他语言，请参考如下

### 1、切换语言

Vion 通过 Locale 组件实现多语言支持，使用`Locale.use`方法切换语言环境，比如切换到英文环境

```js
// 按需引入
import { createApp } from "vue";
import { ActionSheet, Locale } from "vion";
import messages from "vion/es/locale/lang/en-US";

Locale.use("en-US", messages);

createApp()
  .use(ActionSheet);
  .use(Locale);

```

或

```js
// 全量引入Vion
import { createApp } from "vue";
import Vion from "vion";
import messages from "vion/es/locale/lang/en-US";

vion.Locale.use("en-US", messages);

createApp().use(Vion);
```

### 2、语言包

目前 vion 内置了以下几种语言

- 简体中文（zh-CN）
- 繁体中文（zh-Hant）
- 英语（en-US）
- 日语（ja-JP）

如果不能满足业务场景，只需要按照[语言包格式](https://github.com/Nice-PLQ/vion/blob/master/src/locale/lang/zh-CN.js)自定义语言，然后使用`Locale`组件加载即可，如：

```js
import messages from "./lang/fr-FR"; // 自定义的【法语】语言包
Locale.use("fr-FR", messages);
```

### 3、引入多个语言包

```js
import frMessages from "./lang/fr-FR"; // 自定义的【法语】语言包
import trMessages from "./lang/tr-TR"; // 自定义的【土耳其语】语言包

Locale.add("tr-TR", trMessages); // 将语言包加入到集合中
Locale.use("fr-FR", frMessages); // 默认使用法语作为首选语言
```
