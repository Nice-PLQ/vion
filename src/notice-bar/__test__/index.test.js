import { mount } from '@vue/test-utils';
import NoticeBar from '../notice-bar.vue';
import i18n from '../../locale';

describe('NoticeBar', () => {
  test('渲染', () => {
    const wrapper = mount({
      template: `
        <div>
          <v-notice-bar type="warning" right-icon-type="right">网络出现异常，请检查。</v-notice-bar>
          <v-notice-bar type="info" right-text="去设置">这是一段超长的提示文本，超长的部分将会自动隐藏</v-notice-bar>
          <v-notice-bar type="black" right-icon-type="close">你有新的邮件，请及时查阅。</v-notice-bar>
          <v-notice-bar type="gray" right-icon-type="close">你有新的邮件，请及时查阅。</v-notice-bar>
        </div>
      `
    }, {
      global: {
        components: { VNoticeBar: NoticeBar },
        config: {
          globalProperties: {
            $locale: i18n
          }
        }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('事件触发', async () => {
    const wrapper = mount(NoticeBar, {
      props: {
        rightText: '去设置',
        rightIconType: 'close'
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n
          }
        }
      }
    });

    await wrapper.find('.v-notice__setting').trigger('click');
    await wrapper.find('.v-notice__right-icon').trigger('click');
    expect(wrapper.emitted('right-text-click')).toBeTruthy();
    expect(wrapper.emitted('right-icon-click')).toBeTruthy();
  });
});
