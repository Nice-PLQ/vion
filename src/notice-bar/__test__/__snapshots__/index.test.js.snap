// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar 渲染 1`] = `
<div>
  <div
    class="v-notice v-notice_warning"
  >
    <div
      class="v-notice__inner"
    >
      <div
        class="v-notice__hd"
      >
        <div
          class="v-notice__left-icon"
        >
          
          <i
            class="v-icon"
            style="display: inline-block; font-size: 24px; width: 1em; height: 1em; line-height: 1em;"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M44 24C44 12.9543 35.0457 4 24 4C12.9543 4 4 12.9543 4 24C4 35.0457 12.9543 44 24 44C35.0457 44 44 35.0457 44 24ZM22 31V35H26V31H22ZM22 13V29H26V13H22Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          
        </div>
      </div>
      <div
        class="v-notice__bd"
      >
        <span
          class="v-notice__txt"
        >
          
          网络出现异常，请检查。
          
        </span>
      </div>
      <!--v-if-->
      <i
        aria-label="更多"
        class="v-icon v-notice__right-icon"
        role="button"
        style="display: inline-block; font-size: 15px; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M34.9627 24.0099L34.9741 23.9985L18.0035 7.02795L15.1751 9.85638L29.3377 24.019L15.1842 38.1725L18.0126 41.0009L34.9831 24.0303L34.9627 24.0099Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    class="v-notice v-notice_info"
  >
    <div
      class="v-notice__inner"
    >
      <div
        class="v-notice__hd"
      >
        <div
          class="v-notice__left-icon"
        >
          
          <i
            class="v-icon"
            style="display: inline-block; font-size: 24px; width: 1em; height: 1em; line-height: 1em;"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M44 24C44 12.9543 35.0457 4 24 4C12.9543 4 4 12.9543 4 24C4 35.0457 12.9543 44 24 44C35.0457 44 44 35.0457 44 24ZM22 17V13H26V17H22ZM22 35V19H26V35H22Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          
        </div>
      </div>
      <div
        class="v-notice__bd"
      >
        <span
          class="v-notice__txt"
        >
          
          这是一段超长的提示文本，超长的部分将会自动隐藏
          
        </span>
      </div>
      <div
        class="v-notice__ft"
      >
        <span
          class="v-notice__setting"
          role="button"
        >
          去设置
        </span>
      </div>
      <i
        aria-label="更多"
        class="v-icon v-notice__right-icon"
        role="button"
        style="display: inline-block; font-size: 15px; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M34.9627 24.0099L34.9741 23.9985L18.0035 7.02795L15.1751 9.85638L29.3377 24.019L15.1842 38.1725L18.0126 41.0009L34.9831 24.0303L34.9627 24.0099Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    class="v-notice v-notice_black"
  >
    <div
      class="v-notice__inner"
    >
      <div
        class="v-notice__hd"
      >
        <div
          class="v-notice__left-icon"
        >
          
          <i
            class="v-icon"
            style="display: inline-block; font-size: 24px; width: 1em; height: 1em; line-height: 1em;"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M44 24C44 12.9543 35.0457 4 24 4C12.9543 4 4 12.9543 4 24C4 35.0457 12.9543 44 24 44C35.0457 44 44 35.0457 44 24ZM22 17V13H26V17H22ZM22 35V19H26V35H22Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          
        </div>
      </div>
      <div
        class="v-notice__bd"
      >
        <span
          class="v-notice__txt"
        >
          
          你有新的邮件，请及时查阅。
          
        </span>
      </div>
      <!--v-if-->
      <i
        aria-label="关闭"
        class="v-icon v-notice__right-icon"
        role="button"
        style="display: inline-block; font-size: 15px; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M37.4767 13.435L34.6482 10.6066L24.0416 21.2132L13.435 10.6066L10.6066 13.435L21.2132 24.0416L10.6066 34.6482L13.435 37.4767L24.0416 26.8701L34.6482 37.4767L37.4767 34.6482L26.8701 24.0416L37.4767 13.435Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    class="v-notice v-notice_gray"
  >
    <div
      class="v-notice__inner"
    >
      <div
        class="v-notice__hd"
      >
        <div
          class="v-notice__left-icon"
        >
          
          <i
            class="v-icon"
            style="display: inline-block; font-size: 24px; width: 1em; height: 1em; line-height: 1em;"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M44 24C44 12.9543 35.0457 4 24 4C12.9543 4 4 12.9543 4 24C4 35.0457 12.9543 44 24 44C35.0457 44 44 35.0457 44 24ZM22 17V13H26V17H22ZM22 35V19H26V35H22Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </i>
          
        </div>
      </div>
      <div
        class="v-notice__bd"
      >
        <span
          class="v-notice__txt"
        >
          
          你有新的邮件，请及时查阅。
          
        </span>
      </div>
      <!--v-if-->
      <i
        aria-label="关闭"
        class="v-icon v-notice__right-icon"
        role="button"
        style="display: inline-block; font-size: 15px; width: 1em; height: 1em; line-height: 1em;"
      >
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M37.4767 13.435L34.6482 10.6066L24.0416 21.2132L13.435 10.6066L10.6066 13.435L21.2132 24.0416L10.6066 34.6482L13.435 37.4767L24.0416 26.8701L34.6482 37.4767L37.4767 34.6482L26.8701 24.0416L37.4767 13.435Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </i>
    </div>
  </div>
</div>
`;
